src/internalTypes.ts

```diff
@@ -31,19 +46,35 @@ export interface IResponseMock {
     [key: string]: unknown;
 }
 
-export interface IWireMockMockedRequestResponse {
-    id: string;
-    newScenarioState?: string;
-    priority?: number;
-    request: IRequestMock;
-    requiredScenarioState?: string;
-    response: IResponseMock;
-    scenarioName?: string;
-    uuid: number;
+export interface IScenarioGetResponse {
+    scenarios: Array<unknown>;
 }
 
-export interface IWireMockScenario {
-    scenarioName: string;
-    requiredScenarioState: 'Started' | string;
-    newScenarioState?: string;
+export interface IWebhook {
+    name: 'webhook';
+    parameters: {
+        method: Method;
+        url: string;
+        headers?: Record<string, KeyValue>;
+        body?: string;
+        delay?: Record<string, number | string>;
+    };
 }
+
+export type KeyValue = boolean | number | string;
+
+export type Method =
+    | 'ANY'
+    | 'CONNECT'
+    | 'DELETE'
+    | 'GET'
+    | 'HEAD'
+    | 'OPTIONS'
+    | 'PATCH'
+    | 'POST'
+    | 'PUT'
+    | 'TRACE';
+
+export type WebhookBody =
+    | { type: 'JSON'; data: Record<string, unknown> }
```

Comment:
Why is `data` a Record? Isn't it same as`IWebhook.parameters`? If so we can extract that as a type and reuse.

Reply:
wiremock expects webhook body to be provided as `string`. so if your webhook is a `POST` with some body, the expectation is to provide `JSON.stringify` version to wiremock webhook

do have an easier interface for end user, i want to allow them to pass JSON body and we can do the `JSON.stringify` before setting up the stub with wiremock (or the user can pass a string themselves)